src/Build/Logging/TerminalLogger/TerminalLogger.cs

```diff
@@ -250,6 +229,72 @@ internal TerminalLogger(ITerminal terminal)
         _manualRefresh = true;
     }
 
+    /// <summary>
+    /// Private constructor invoked by static factory.
+    /// </summary>
+    internal TerminalLogger(LoggerVerbosity verbosity, uint? originalConsoleMode) : this()
+    {
+        Verbosity = verbosity;
+        _originalConsoleMode = originalConsoleMode;
+    }
+
+    /// <summary>
+    /// Creates a Terminal logger if possible, or a Console logger.
+    /// </summary>
+    /// <param name="args">Command line arguments for the logger configuration. Currently, only 'tl|terminallogger' and 'v|verbosity' are supported right now.</param>
+    public static ILogger CreateTerminalOrConsoleLogger(string[]? args = null)
+    {
+        (bool supportsAnsi, bool outputIsScreen, uint? originalConsoleMode) = NativeMethodsShared.QueryIsScreenAndTryEnableAnsiColorCodes();
+
+        return CreateTerminalOrConsoleLogger(args, supportsAnsi, outputIsScreen, originalConsoleMode);
+    }
+
+    internal static ILogger CreateTerminalOrConsoleLogger(string[]? args, bool supportsAnsi, bool outputIsScreen, uint? originalConsoleMode)
+    {
+        LoggerVerbosity verbosity = LoggerVerbosity.Normal;
+        string tlArg = string.Empty;
+        string? verbosityArg = string.Empty;
+
+        if (args != null)
+        {
+            string argsString = string.Join(" ", args);
+
+            MatchCollection tlMatches = Regex.Matches(argsString, @"(?:/|-|--)(?:tl|terminallogger):(?'value'on|off)", RegexOptions.IgnoreCase);
+            tlArg = tlMatches.OfType<Match>().LastOrDefault()?.Groups["value"].Value ?? string.Empty;
+
+            MatchCollection verbosityMatches = Regex.Matches(argsString, @"(?:/|-|--)(?:v|verbosity):(?'value'\w+)", RegexOptions.IgnoreCase);
+            verbosityArg = verbosityMatches.OfType<Match>().LastOrDefault()?.Groups["value"].Value;
+        }
+
+        verbosityArg = verbosityArg?.ToLowerInvariant() switch
+        {
+            "q" => "quiet",
+            "m" => "minimal",
+            "n" => "normal",
+            "d" => "detailed",
+            "diag" => "diagnostic",
+            _ => verbosityArg,
+        };
+
+        if (Enum.TryParse(verbosityArg, true, out LoggerVerbosity parsedVerbosity))
+        {
+            verbosity = parsedVerbosity;
+        }
+
+        bool isDisabled =
+            tlArg.Equals("on", StringComparison.InvariantCultureIgnoreCase) ? false :
+            tlArg.Equals("off", StringComparison.InvariantCultureIgnoreCase) ? true :
+            (Environment.GetEnvironmentVariable("MSBUILDTERMINALLOGGER") ?? string.Empty).Equals("off", StringComparison.InvariantCultureIgnoreCase);
```

Comment:
this is different than creating Terminal logger in Xmake 😩 - there it recognizes "false" and "0" as well. (it's a bit silly part of Xmake) https://github.com/dotnet/msbuild/blob/072c659ff4d49bdf70069cd36df6dc9e44719432/src/MSBuild/XMake.cs#L2966

Also could this code live in Traits?

Reply:
I'm probably blind, but the `NormalizeIntoBooleanValues()` is testing the variable for `on` and `off` values :)

Reply:
oops I meant to link the whole method processing tl argshttps://github.com/dotnet/msbuild/blob/072c659ff4d49bdf70069cd36df6dc9e44719432/src/MSBuild/XMake.cs#L2826 , normalize is just one part of it


the logic is basically:
1. try getting -tl: value from console or from the env var
2. convert "on" to "true", "off" to "false"
3. parse a boolean from a string -> the actual info whether to use terminal logger

(at this point I am confused what it'll do for 0 and 1, but for true and false it'll definitely set the tl on or off)

Reply:
Yes, it recognizes `true` and `false` values, although I'm not sure if it's intentional :) With all these local functions, it's difficult to read the code sequentially... Anyway, I added support for `false` value and I will switch to `Traits` in the second phase. I don't know if we still need to support `MSBUILDLIVELOGGER` variable in .NET 10.